sidecar/src/controller.rs

```diff
@@ -0,0 +1,76 @@
+#![allow(dead_code)] // TODO remove when it is implemented
+
+use thiserror::Error;
+use tokio::select;
+use tokio::sync::oneshot::Receiver;
+use tokio::time::{Instant, Interval};
+
+use crate::config::Config;
+
+/// Sidecar operator controller
+#[derive(Debug)]
+pub(crate) struct Controller {
+    /// Sidecar operator config
+    config: Config,
+    /// Check interval
+    check_interval: Interval,
+}
+
+/// All possible errors
+#[derive(Error, Debug, PartialEq)]
+pub(crate) enum Error {
+    /// Graceful shutdown error
+    #[error("graceful shutdown")]
```

Comment:
Don't use "graceful shutdown" as the error message. A client may not understand how graceful it is. IMO, "Operator has been shutdown" or something like that would be better.

Reply:
done